proxy/coco/src/peer.rs

```diff
@@ -167,4 +196,68 @@ impl Peer {
             sender.clone().send(TimeoutEvent::SyncPeriod).await.ok();
         });
     }
+
+    /// Request subroutine.
+    async fn request(
```

Comment:
To keep the core peer impl light, this could follow the precedence of `announcement` and `sync` and live in `coco::peer::request`.